### PR TITLE
fix: table version changed for crud

### LINKED ISSUE #29567: [Bug]: Update the table version to v2 for generate CRUD functionality
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Update the table widget version to v2 for generate crud functionality which is using v1 version right now

### Steps To Reproduce

1. Create a datasource preferably mysql, postgres, gsheet 
2. Click on generate page button to create the crud operation pages
3. Observe the table, its different what current table is present in the widgets panel.

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Both

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
Following are the changes that this PR introduces :
- Earlier the generate crud operation was using Table widget version v1 with the help of `CRUD-DB-Table-Template-Application.json` which is now changed to Table widget version v2 after this change
- Test cases that were present using the generate crud have now been changed to support Table v2 version.

#### PR fixes following issue(s)
Fixes #29567
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Tests**
	- Updated Cypress end-to-end test scripts to align with the version change from "v1" to "v2" for various modules and components.
	- Enhanced data validation and refined deployment workflow in test cases.
	- Adjusted test scripts to account for updated control flow and interaction with table components.

- **Refactor**
	- Modified table column rendering logic to prioritize new column IDs.

- **Chores**
	- Standardized the use of "click" actions in Cypress support files to improve consistency and reliability in automated tests.

### CODE REVIEW

**Comment by AmanAgarwal041** [comment]
This is a draft pr to try out the updation of table version for generate crud functionality after looking into the previously closed issue as well as pr and some conversation behind that decision. Here are my observations :
- There is just the backend change in `app/server/appsmith-server/src/main/resources/CRUD-DB-Table-Template-Application.json` file to enable the usage of table version v2 for generate crud functionality
- We need to look into the different failing test cases that are still using table v1 to verify the working application
- We need to remove the table version v1 specific test cases from the repo

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -721,7 +721,7 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
         ) {
           // Maintain original columnOrder and keep new columns at the end
           let newColumnOrder = _.intersection(columnOrder, newColumnIds);
-          newColumnOrder = _.union(newColumnOrder, newColumnIds);
+          newColumnOrder = _.union(newColumnIds, newColumnOrder);
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/src/widgets/TableWidgetV2/widget/index.tsx:724]: This change is made temporarily to check if cypress tests are passing if we fix the ordering issue. I will be reverting this commit once we are sure on cypress tests

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Pull request merged**